core/lib/zksync_core/src/commitment_generator/mod.rs

```diff
@@ -210,11 +214,35 @@ impl CommitmentGenerator {
             }
             state_diffs.sort_unstable_by_key(|rec| (rec.address, rec.key));
 
+            let kzg_settings = load_kzg_settings();
```

Comment:
load_kzg_settings is heavy - maybe move inside the 'if' (so that we don't invoke it in non-1.4.2 flow?)

(unless this is done on purpose to test things - but then let's mention it in the comment)

Reply:
It isn't heavy due to it being an once_cell, but moving it inside `if` certainly makes sense.